build_script/jfrog.gradle

```diff
@@ -0,0 +1,61 @@
+/*
+   Copyright (c) 2021 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+/**
+ * The purpose of this script is to configure the following tasks for a sub-project:
+ * - artifactoryPublish: publishes all artifacts for the "release" publication to JFrog; depends on...
+ *   - generatePomFileForReleasePublication: defined in publications.gradle
+ *   - assertArtifactsExist: defined in publications.gradle
+ *   - assertBintrayCredentialsExist: asserts that JFrog credentials are present as environment variables
+ */
+apply plugin: 'com.jfrog.artifactory' // https://www.jfrog.com/confluence/display/rtf/gradle+artifactory+plugin
+
+final String jfrogUserEnv = 'JFROG_USER'
+final String jfrogKeyEnv = 'JFROG_KEY'
+
+def jfrogUser = System.getenv(jfrogUserEnv)
+def jfrogKey = System.getenv(jfrogKeyEnv)
+
+// Configure the "artifactoryPublish" task for the "release" publication.
+artifactory {
+  contextUrl = 'https://linkedin.jfrog.io/artifactory'
+  clientConfig.setIncludeEnvVars(false)
+
+  publish {
+    repository {
+      repoKey = 'test-repo' // The Artifactory repository key to publish to // TODO: change this to "pegasus" once this has been tested in CI
+      username = jfrogUser // The publisher user name
+      password = jfrogKey // The publisher password
+    }
+
+    defaults {
+      publications(publishing.publications.release)
+    }
+  }
+}
+
+// Utility task to ensure that we aren't attempting a publish without providing JFrog credentials.
+task assertJFrogCredentialsExist() {
+  doLast {
+    if (!jfrogUser || !jfrogKey) {
+      throw new GradleException(
+          "Cannot perform JFrog upload. Missing '${jfrogUserEnv}' or '${jfrogKeyEnv}' environment variable.")
```

Comment:
When printed, will we see `${jfrogUserEnv}` or `JFROG_USER` or <<...empty...>>?
And if this message does come up, how do we get to the environment and set these variables?

Reply:
This just prints the name of the variable (e.g. `JFROG_USER`). This can be resolved via `export JFROG_USER=foo` in bash before running the Gradle task. However, this is normally done in CI so the env variables are set in the Travis config.

Reply:
I believe the user/key is configured in Travis for all projects under linkedin. Is that right? Might be a good idea to add a comment here to explain how they are configured.